### PR TITLE
Fix Home Owners validation and Review page

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* /bcgov/entity#13710

*Description of changes:*
- Fix Home Owner step validation
- Fix Review page (multiple bugs)
- Add unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by cameron-eyds — APPROVED** [review]
Looks good, nice job 👍

**Review by doug-lovett — APPROVED** [review]
LGTM.

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 1]
```diff
@@ -310,11 +304,24 @@ export function useHomeOwners (isPerson: boolean = false, isEditMode: boolean =
...
+      isHomeOwnersStepValid = !hasEmptyGroup.value
+      isHomeOwnersStepValid = !getTotalOwnershipAllocationStatus().hasTotalAllocationError
+      isHomeOwnersStepValid = !getTotalOwnershipAllocationStatus().hasMinimumGroupsError
+    } else {
+      // must have at least one owner with proper id
+      isHomeOwnersStepValid = !!getMhrRegistrationHomeOwners.value.find(owner => owner.id)
+    }
+    setValidation(MhrSectVal.HOME_OWNERS_VALID, MhrCompVal.OWNERS_VALID, isHomeOwnersStepValid)
+  })
+
```
Comment (ChasBoucher) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts:324]: Looks Good.

**Review by ChasBoucher — COMMENTED** [review]
I did some testing and only one condition does not change stepper validation:

When there are groups and one of the groups does not have owners still shows as valid.

**Pull request merged**